src/crypto/Crypto.cpp

```diff
@@ -19,326 +19,288 @@
+
+        QByteArray data = plainText;
+        SymmetricCipher twofish;
+        if (!twofish.init(SymmetricCipher::Twofish_CBC, SymmetricCipher::Encrypt, key, iv)) {
+            raiseError(twofish.errorString());
+            return false;
+        }
+        if (!twofish.process(data)) {
+            raiseError(twofish.errorString());
+            return false;
+        }
+        if (data != cipherText) {
+            raiseError("Twofish encryption mismatch.");
+            return false;
+        }
+
+        if (!twofish.init(SymmetricCipher::Twofish_CBC, SymmetricCipher::Decrypt, key, iv)) {
+            raiseError(twofish.errorString());
+            return false;
+        }
+        if (!twofish.process(data)) {
+            raiseError(twofish.errorString());
+            return false;
+        }
+        if (data != plainText) {
+            raiseError("Twofish encryption mismatch.");
+            return false;
+        }
 
-bool Crypto::testSalsa20()
-{
-    QByteArray salsa20Key = QByteArray::fromHex("F3F4F5F6F7F8F9FAFBFCFDFEFF000102030405060708090A0B0C0D0E0F101112");
-    QByteArray salsa20iv = QByteArray::fromHex("0000000000000000");
-    QByteArray salsa20Plain = QByteArray::fromHex("00000000000000000000000000000000");
-    QByteArray salsa20Cipher = QByteArray::fromHex("B4C0AFA503BE7FC29A62058166D56F8F");
-    bool ok;
-
-    SymmetricCipher salsa20Stream(SymmetricCipher::Salsa20, SymmetricCipher::Stream, SymmetricCipher::Encrypt);
-    if (!salsa20Stream.init(salsa20Key, salsa20iv)) {
-        raiseError(salsa20Stream.errorString());
-        return false;
-    }
-
-    QByteArray salsaProcessed = salsa20Stream.process(salsa20Plain, &ok);
-    if (!ok) {
-        raiseError(salsa20Stream.errorString());
-        return false;
-    }
-    if (salsaProcessed != salsa20Cipher) {
-        raiseError("Salsa20 stream cipher mismatch.");
-        return false;
+        return true;
     }
 
-    return true;
-}
+    bool testSalsa20()
+    {
+        QByteArray salsa20Key = QByteArray::fromHex("F3F4F5F6F7F8F9FAFBFCFDFEFF000102030405060708090A0B0C0D0E0F101112");
+        QByteArray salsa20iv = QByteArray::fromHex("0000000000000000");
+        QByteArray salsa20Plain = QByteArray::fromHex("00000000000000000000000000000000");
+        QByteArray salsa20Cipher = QByteArray::fromHex("B4C0AFA503BE7FC29A62058166D56F8F");
+
+        QByteArray data = salsa20Plain;
+        SymmetricCipher salsa20Stream;
+        if (!salsa20Stream.init(SymmetricCipher::Salsa20, SymmetricCipher::Encrypt, salsa20Key, salsa20iv)) {
+            raiseError(salsa20Stream.errorString());
+            return false;
+        }
+        if (!salsa20Stream.process(data)) {
+            raiseError(salsa20Stream.errorString());
+            return false;
+        }
+        if (data != salsa20Cipher) {
+            raiseError("Salsa20 stream cipher mismatch.");
+            return false;
+        }
```

Comment:
Stream ciphers are bidirectional by design you wouldn't technically be proving anything, but we can add the check.

Reply:
You would prove that what comes out of the encryption pass is actually the result of a stream cipher. But, yeah, we check for the result already, so it would be more of a double check that the test ist correct. You may also leave it out.